src/Workspaces/Remote/ServiceHub/Shared/ServiceHubServiceBase.cs

```diff
@@ -82,6 +91,37 @@ protected RoslynServices RoslynServices
             }
         }
 
+        protected bool IsDisposed => ((IDisposableObservable)_rpc).IsDisposed;
+
+        protected void StartService()
+        {
+            _rpc.StartListening();
+        }
+
+        protected Task<TResult> InvokeAsync<TResult>(
+            string targetName, IReadOnlyList<object> arguments, CancellationToken cancellationToken)
```

Comment:
consider: args and cancellationToken params could be optional (to avoid passing CancellationToken.None for example)

Reply:
@shyamnamboodiripad I would definitely *not* want that.  It seems like that could easily lead to bugs where someone forgot to pass int eh cancellation token and then ended up using too many resources when something was actually canceled.

Reply:
our pattern is that internal API must be all explicit. that I think you probably need to create extension method in your side.

Reply:
Makes sense - sounds good.